packages/workflow/src/utils.ts

```diff
@@ -1,12 +1,15 @@
 /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-argument */
-export const deepCopy = <T>(source: T): T => {
+export const deepCopy = <T>(source: T, hash = new WeakMap()): T => {
 	let clone: any;
 	let i: any;
 	const hasOwnProp = Object.prototype.hasOwnProperty.bind(source);
-	// Primitives & Null
-	if (typeof source !== 'object' || source === null) {
+	// Primitives & Null & Function
+	if (typeof source !== 'object' || source === null || source instanceof Function) {
```

Comment:
functions are also objects and can form circular dependencies. should we be handling that case?

Reply:
That's why they're just returned as primitives. Those wouldn't make sense to copy anyways because what matters is their return value after the invocation and not themselves. During deepcopy no functions are invoked

Reply:
Ok, I see your point now. It would be very bad practice to define a function and then give them own properties
Like: 
```ts
const x = () => {}
x.myProp = 'whatever'